mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxLifecycleAwareFlow.kt

```diff
@@ -0,0 +1,70 @@
+@file:Suppress("EXPERIMENTAL_API_USAGE")
+
+package com.airbnb.mvrx
+
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.channelFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+
+/**
+ * Similar to LifecycleScope.launchWhenStarted but cancels the block when the owner is stopped.
+ * This function suspends until the next time the owner is stopped.
+ */
+private suspend fun LifecycleOwner.launchWhenStartedAndCancelWhenStopped(block: suspend () -> Unit) {
+    var parentJob = SupervisorJob()
+    lifecycle.addObserver(object : LifecycleEventObserver {
+        override fun onStateChanged(source: LifecycleOwner, event: Lifecycle.Event) {
+            when (event) {
+                Lifecycle.Event.ON_START -> lifecycleScope.launch(parentJob) { block() }
+                Lifecycle.Event.ON_STOP -> {
+                    lifecycle.removeObserver(this)
```

Comment:
if added after onStop this is never removed, not sure if that is a problem

Reply:
@elihart I moved the remove to ON_DESTROY where I think it should have been